### PR TITLE
Fixed code issue from compat with new numpydoc

### LINKED ISSUE #26187: Doc Check Failures
This just started showing up in the CI failures today:

```sh
Traceback (most recent call last):
  File "ci/../scripts/validate_docstrings.py", line 991, in <module>
    args.ignore_deprecated))
  File "ci/../scripts/validate_docstrings.py", line 891, in main
    result = validate_all(prefix, ignore_deprecated)
  File "ci/../scripts/validate_docstrings.py", line 845, in validate_all
    doc_info = validate_one(func_name)
  File "ci/../scripts/validate_docstrings.py", line 801, in validate_one
    errs, wrns, examples_errs = get_validation_data(doc)
  File "ci/../scripts/validate_docstrings.py", line 749, in get_validation_data
    if not doc.see_also:
  File "ci/../scripts/validate_docstrings.py", line 477, in see_also
    in self.doc['See Also'])
  File "ci/../scripts/validate_docstrings.py", line 476, in <genexpr>
    for name, desc, _
ValueError: not enough values to unpack (expected 3, got 2)
```

I think it's an issue with upgrading to numpydoc 0.9 but will look in more detail and confirm

### PR DESCRIPTION
**Pull request by WillAyd**
- [X] closes #26187
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: scripts/validate_docstrings.py [hunk 1]
```diff
@@ -473,8 +473,7 @@ def parameter_desc(self, param):
     @property
     def see_also(self):
         return collections.OrderedDict((name, ''.join(desc))
-                                       for name, desc, _
-                                       in self.doc['See Also'])
+                                       for name, desc in self.doc['See Also'])
```
Comment (jreback) [inline scripts/validate_docstrings.py]: prob want to put a ``numpydoc>=0.9.0`` in environment.yml

alt you might be able to 

``for name, desc, * in self.doc[....]``

File: scripts/validate_docstrings.py [hunk 2]
```diff
@@ -472,9 +472,12 @@ def parameter_desc(self, param):
 
     @property
     def see_also(self):
-        return collections.OrderedDict((name, ''.join(desc))
-                                       for name, desc, _
-                                       in self.doc['See Also'])
+        result = collections.OrderedDict()
+        for funcs, desc in self.doc['See Also']:
```
Comment (jreback) [inline scripts/validate_docstrings.py]: could write this as a list comprehension but not a big deal

**Comment by rgommers** [comment]
Hmm, this seemed to rely on some `numpydoc` implementation details, we at least don't consider `numpydoc.docscrape.NumpyDocString` public. PR LGTM.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jreback** [comment]
thanks @WillAyd

**Pull request merged**